.github/workflows/publish-release-build.yml

```diff
@@ -17,7 +17,7 @@ jobs:
       - uses: actions/setup-java@v3
         with:
           distribution: 'zulu'
-          java-version: 19
+          java-version: 20
```

Comment:
To clear things out. This is now irrelevant to how the build will behave. Java version is controlled via Gradle toolchains and setting it to Java 20 only improves CI performance since it now doesn't have to be downloaded and installed on every run.
With toolchains the build will behave exactly the same way on CI & on local builds

Reply:
>Please add a comment that the latest java version needs to be used for compilation.

to clarify, do you mean to the changelog entry? Or should add an inline comment here? 

>the latest java version needs to be used for compilation.

To be extra clear of how toolchains work: even if you have only Java 11 installed and set as default, Gradle should automatically (via [foojay toolchains](https://github.com/gradle/foojay-toolchains)) fetch the Java version required by the project. So there should be no extra effort required to configure local environment. If you have _any_ Java installed and Gradle is able to run on it, it will compile both projects (`/build-logic` and `/`) using Java version prividet in `jvmToolchain()` call (which is currently set to Java 20).